src/Xamarin.Android.Build.Tasks/Tasks/RegisterAssemblyStoreState.cs

```diff
@@ -0,0 +1,26 @@
+using System;
+using System.IO;
+using System.Collections.Generic;
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Microsoft.Android.Build.Tasks;
+
+namespace Xamarin.Android.Tasks
+{
+	public class RegisterAssemblyStoreState : AndroidTask
+	{
+		public const string RegisteredObjectKey = ".:RegisterAssemblyStoreState_Key:.";
+
+		public override string TaskPrefix => "RASS";
+
+		[Required]
+		public bool UseAssemblyStore { get; set; }
```

Comment:
I think some of the issues with incremental builds are due to this task and usage of `RegisterTaskObject`.

Can we instead do something like?

1. Delete the `<RegisterAssemblyStoreState/>` task
2. Add `UseAssemblyStore` to `<BuildApk/>`. Don't make it required.
3. Sometime in the future we could update monodroid, but I don't think we even want this value used for Fast Deploy scenarios?

https://github.com/xamarin/monodroid/blob/9b3a37af1102f56ccaf449864f18a7d2a78e9682/tools/msbuild/Xamarin.Android.Common.Debugging.targets#L238-L268

Maybe we don't ever need step 3 if this isn't meant to work for Fast Deployment.

I think that might solve the bugs here and be less code.

Reply:
I added this task today to avoid precisely the problem you're describing. `GenerateJavaStubs` registers an app config object which, before I added the above code, had a boolean property telling `BuildApk` whether to use assembly stores. It worked fine but not with `dotnet run` which appears to build the app twice - first the build and second the packaging. In the build phase, app config had `true` for the `UseAssemblyStore` flag, and that value was placed in the generated native assembly code - the runtime would expect to find assembly store blobs in the apk. However, in the packaging phase of `dotnet run` `GenerateJavaStubs` was skipped and so the app config object was never registered and `BuildApk` packaged **assemblies** in the apk instead of the blobs, resulting in the native runtime aborting execution when it could not find the blobs.